go/libraries/doltcore/schema/constraint.go

```diff
@@ -92,15 +92,47 @@ func IndexOfConstraint(constraints []ColConstraint, constraintType string) int {
 
 // ColConstraintsAreEqual validates two ColConstraint slices are identical.
 func ColConstraintsAreEqual(a, b []ColConstraint) bool {
-	if len(a) != len(b) {
+	// TODO: do something with maps to avoid all duplicate constraint issues, and order issues too
```

Comment:
I would revert this change. We should have a well-formed schema by the time we're working with it. Let's apply the validation/cleaning earlier